consensus/reactor.go

```diff
@@ -196,6 +200,19 @@ func (r *Reactor) OnStart() error {
 // blocking until they all exit, as well as unsubscribing from events and stopping
 // state.
 func (r *Reactor) OnStop() {
+
+	// If the node is committing the new block, wait until it finished!
+
+	if r.state.GetRoundState().Step == cstypes.RoundStepCommit {
+		select {
+		case <-r.newHeightCh:
+		case <-time.After(3 * time.Second):
+			r.Logger.Error("the 3 secs block commit waiting timeout! The block commit might be failed!")
+		}
+	}
```

Comment:
I don't think you need to create a new event. We're just wanting to wait till `RoundStepCommit` is done, so we should be able to use `EventNewRoundStep` to monitor if that happens or not

Reply:
Also I'm wondering if it's better to do this in the `OnStop` func within `State` rather than the reactor

Reply:
Right, I will move it into the state.OnStop(). Thanks for the suggestion.